src/wp-admin/install.php

```diff
@@ -299,119 +147,286 @@ function display_setup_form( $error = null ) {
+	echo '	</tr>' . "\n";
+	// Admin username.
+	echo '	<tr>' . "\n";
+	echo '		<th scope="row"><label for="user_login">' . __('Admin Username') . '</label></th>' . "\n";
+	echo '		<td>';
+	// Desired user already exists?
+	if ( $user_table ) {
+		_e('User(s) already exists.');
+		echo '<input name="user_name" type="hidden" value="admin" />';
+	} else { // ...no?
+		echo '<input name="user_name" type="text" id="user_login" size="25" value="' . esc_attr( sanitize_user( $user_name, true ) ) . '" />';
+		echo '<p class="description" style="max-width:300px;">' . __( 'Letters, numbers, spaces, underscores, hyphens, periods, and the @ symbol only. Usernames cannot be changed.' ) . '</p>';
+	}
+	echo '</td>' . "\n";
+	echo '	</tr>' . "\n";
+	// Non-existing user? Great. Add the admin password fields.
+	if ( ! $user_table ) {
+		// Create a random password to suggest.
+		$initial_password = isset( $_POST['admin_password'] ) ? stripslashes( $_POST['admin_password'] ) : wp_generate_password( 18 );
+		// Admin password.
+		echo '	<tr class="form-field form-required user-pass1-wrap">' . "\n";
+		echo '		<th scope="row">' . "\n";
+		echo '			<label for="pass1">' . __( 'Admin Password' ) . '</label>' . "\n";
+		echo '		</th>' . "\n";
+		echo '		<td>' . "\n";
+		echo '			<div>' . "\n";
+		echo '				<input type="password" name="admin_password" id="pass1" class="regular-text" autocomplete="off" data-reveal="1" data-pw="' . esc_attr( $initial_password ) . '" aria-describedby="pass-strength-result" />' . "\n";
+		echo '				<button type="button" class="button wp-hide-pw hide-if-no-js" data-start-masked="' . (int) isset( $_POST['admin_password'] ) . '" data-toggle="0" aria-label="' . esc_attr( 'Hide password' ) . '">' . "\n";
+		echo '					<span class="dashicons dashicons-hidden"></span>' . "\n";
+		echo '					<span class="text">' . __( 'Hide' ) . '</span>' . "\n";
+		echo '				</button>' . "\n";
+		echo '				<div id="pass-strength-result" aria-live="polite"></div>' . "\n";
+		echo '			</div>' . "\n";
+		echo '		</td>' . "\n";
+		echo '	</tr>' . "\n";
+		// Repeat password.
+		echo '	<tr class="form-field form-required user-pass2-wrap hide-if-js">' . "\n";
+		echo '		<th scope="row">' . "\n";
+		echo '			<label for="pass2">' . __( 'Repeat Password' );
+		echo '				<span class="description">' . __( '(required)' ) . '</span>' . "\n";
+		echo '			</label>' . "\n";
+		echo '		</th>' . "\n";
+		echo '		<td>' . "\n";
+		echo '			<input name="admin_password2" type="password" id="pass2" autocomplete="off" />' . "\n";
+		echo '		</td>' . "\n";
+		echo '	</tr>' . "\n";
+		// Weak password confirmation.
+		echo '	<tr class="pw-weak">' . "\n";
+		echo '		<th scope="row">' . __( 'Confirm Password' ) . '</th>' . "\n";
+		echo '		<td>' . "\n";
+		echo '			<label>' . "\n";
+		echo '			<input type="checkbox" name="pw_weak" class="pw-checkbox" />' . __( 'Confirm use of weak password' );
+		echo '			</label>' . "\n";
+		echo '		</td>' . "\n";
+		echo '	</tr>' . "\n";
```

Comment:
I like this approach to PHP/HTML soup.  The `echo` statements and the indentation provide visual alignment in the source code, and the `"\n"` makes the HTML more readable after it gets to the browser.  It's often necessary to inspect and make sure this is working as intended, so I think it's worth including.

Reply:
Great! I was dreading the possibility of a request to revert it back. I'm not a fan of a ton of echos, but the points you make are exactly why I took this approach. 👍